hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/ReplicationSupervisor.java

```diff
@@ -96,13 +100,19 @@ public ReplicationSupervisor(
             new ThreadFactoryBuilder().setDaemon(true)
                 .setNameFormat("ContainerReplicationThread-%d")
                 .build()),
-        clock);
+        clock, maxQueueSize);
   }
 
   /**
    * Queue an asynchronous download of the given container.
    */
   public void addTask(AbstractReplicationTask task) {
+    if (getQueueSize() >= maxQueueSize) {
```

Comment:
I think here we should use `getTotalInFlightReplications()` instead of `getQueueSize()` as it will include running tasks as well as those queued.